feat: add support for detached head

thatlittleboy opened the pull request:
This PR adds support back for detached head commits which takes this form (in English)

```
# HEAD detached from 2e87307
```
in the COMMIT_EDITMSG file. (this was originally supported by [the-mikedavis/tree-sitter-git-commit](https://github.com/the-mikedavis/tree-sitter-git-commit/blob/main/grammar.js#L197), so most credits go to Mike).

---

Log:
* I've not yet finished adding all the different locales, as mentioned in #9 I've not yet worked out what languages are supported by your parser yet...).
* Also, do you have any advice on obtaining the proper git messages for the different languages? At the moment, I'm just manually changing the locales on my laptop and copying what I see in the gitmessage, but I'm not sure if they are 100% right..

gbprod commented:
For the moment, it seems great!

You can find all locales here : https://github.com/git/git/tree/master/po

To build and test, you also can use `npm run build` and `npm run test`.

Thanks!

thatlittleboy commented:
Got it, thanks for the feedback! I'll work on adding the other languages soon as well once we sort out #13.

thatlittleboy commented:
I've added support for the 17 languages in the README. (and ordered in the same order as the languages in the `README.md`, i.e. alphabetically except for English).

I've also added tests and ran `npm run test` successfully.

---

**Note**: I've only added the "HEAD detached at %s" variant, not the "HEAD detached from %s" variant. Two reasons: (1) I've not yet encountered "from" variant in my own tests, and (2) Mike's `tree-sitter-git-commit` also does not support this.

thatlittleboy commented:
I'll wait for your #17  to be merged in first, then I'll add the bg translation alongside this PR as well.

gbprod commented:
Nice job!
Thanks
